flake.nix

```diff
@@ -1,38 +1,113 @@
-# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io/>
+# SPDX-FileCopyrightText: 2023 Serokell <https://serokell.io/>
 #
 # SPDX-License-Identifier: MPL-2.0
-
 {
-  description = "Get your UTF-8 IO right on the first try";
-
   nixConfig = {
     flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
   };
 
   inputs = {
-    nixpkgs.url = "github:serokell/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    flake-compat = {
+      flake = false;
+    };
     haskell-nix = {
       inputs.hackage.follows = "hackage";
       inputs.stackage.follows = "stackage";
+      url = "github:input-output-hk/haskell.nix";
```

Comment:
Hmm, is there a specific reason why it uses a revision different from what is specified in our flake registry? Ideally, we'd like to reduce the amount of used haskell.nix revisions, because each revision produces a lot of entries in `/nix/store` consuming disk space and increasing cache.

If the revision specified in https://github.com/serokell/flake-registry/blob/master/flake-registry.json doesn't work for some reason, it should be updated